buildenv/jenkins/jobs/infrastructure/artifactory_cleanup.groovy

```diff
@@ -0,0 +1,131 @@
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+*******************************************************************************/
+import java.util.Date
+
+/**
+* This script deletes old artifacts off the Artifactory server 
+* 
+* Parameters: 
+*   JOB_TYPE: Choice
+*             Expected values: one of the following
+*             Time,                // This tells the script to delete all artifacts that are older that were created over x amount of days ago
+*             Number of Builds     // This tells the script to keep x amount of artifacts from a specific project
+*   JOB_TO_CHECK: String – The folder to keep x amount of builds in 
+*   ARTIFACTORY_SERVER: String - The artifactory server to clean up
+*   ARTIFACTORY_REPO: String - The artifactory repo to clean up
+*   ARTIFACTORY_NUM_ARTIFACTS: String - How many artifacts to keep in artifactory
+*   ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS: String - How many days the artifacts can stay in artifactory (if any are exceeded, they are deleted)
+**/
+
+timestamps {
+    node('worker') {
+        checkout scm
+        def variableFile = load 'buildenv/jenkins/common/variables-functions'
+        variableFile.parse_variables_file()
+        variableFile.set_artifactory_config()
+
+        def ARTIFACTORY_SERVER = sever.getUrl()
+        def ARTIFACTORY_REPO = params.ARTIFACTORY_REPO ? params.ARTIFACTORY_REPO : env.ARTIFACTORY_REPO
+        def JENKINS_DOMAIN = params.JENKINS_DOMAIN ? params.JENKINS_DOMAIN : ''
+        def ARTIFACTORY_NUM_ARTIFACTS = params.ARTIFACTORY_NUM_ARTIFACTS ? params.ARTIFACTORY_NUM_ARTIFACTS : env.ARTIFACTORY_NUM_ARTIFACTS
+        def ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS = params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS ? params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS : env.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS
+
+        if (params.JOB_TYPE == 'Time'){
+            cleanupTime(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS)
+        } else if (params.JOB_TYPE == 'Number of Builds'){
+            if (params.JOB_TO_CHECK){
+                cleanupBuilds(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, JENKINS_DOMAIN, params.JOB_TO_CHECK, ARTIFACTORY_NUM_ARTIFACTS)
+            } else {
+                error 'Please input a job to cleanup'
```

Comment:
Can't we get all the folders via the API? This would help with the possibility of layout change. We've already changed how we're laying out the folders since you wrote this.